### ISSUE TITLE
Gets very confused about dynamic method dispatch

### ISSUE THREAD

**Issue by leonerd** [leonerd/tree-sitter-perl-LEGACY#12]
Syntax such as

```perl
$object->${\"a_dynamic_$method"}( @args )
```

causes mass upset. The parser gets upset somewhere before the contents of the quoted string, then sees the closing quote as actually being an open, and much confusion reigns. See screenshot.
![perl-treesitter-dynamic-method](https://user-images.githubusercontent.com/350941/155822992-8abc02d0-ed53-4382-b620-3b4d161ccbd0.png)

**Comment by rabbiveesh** [comment]
lol, it  for sure is assuming the opening quote is escaped. Does this get a new LOLWUT tag?

**Comment by rabbiveesh** [comment]
nope, i was wrong. The issue is that it doesn't recognize scalarref literals anywhere in the grammar.

Should be straightforward (aside from renaming some things for clarity)
